homeassistant/components/modbus/sensor.py

```diff
@@ -89,6 +90,7 @@ def number(value: Any) -> Union[int, float]:
                 vol.Optional(CONF_SCALE, default=1): number,
                 vol.Optional(CONF_SLAVE): cv.positive_int,
                 vol.Optional(CONF_STRUCTURE): cv.string,
+                vol.Optional(CONF_STRUCTURE_INDEX, default=0): cv.positive_int,
```

Comment:
We no longer allow new integrations to use a platform config section or old integrations to do changes to the platform config section. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md.

Please refactor this integration to have the config schema in the base integration module and the config yaml section under the top level integration domain key.